File: C#/0566. Reshape the Matrix.cs
public class Solution {
    public int[,] MatrixReshape(int[,] nums, int r, int c) {
        int row=nums.GetLength(0);    
        int column=nums.GetLength(1);
        if(row*column!=r*c){
            return nums;
        }
        
        int[,] newnums=new int[r,c];
        int newrow=0;
        int newcolumn=0;
        for (int row1=0;row1<row;row1++){
            for(int column1=0;column1<column;column1++){
                if (newcolumn==c){
                    newcolumn=0;
                    newrow+=1;
                }
                newnums[newrow,newcolumn]=nums[row1,column1];
                newcolumn+=1;
            }
        }
        return newnums;
    }
}

File: archives/20190519python/0566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, nums: List[List[int]], r: int, c: int) -> List[List[int]]:
        if len(nums)*len(nums[0])!=r*c:
            return nums
        rep=[[0]*c for i in range(r)]
        
        cur=[]
        for row in range(len(nums)):
            for col in range(len(nums[0])):
                cur.append(nums[row][col])
        
        
        for row in range(len(rep)):
            for col in range(len(rep[0])):
                rep[row][col]=cur[row*len(rep[0])+col]
        
        return rep
        
